README.md

```diff
@@ -48,8 +48,8 @@ The simdjson library is easily consumable with a single .h and .cpp file.
    ```c++
    #include "simdjson.h"
    int main(void) {
-     simdjson::document::parser parser;
-     simdjson::document::element tweets = parser.load("twitter.json");
+     simdjson::dom::parser parser;
```

Comment:
I know that "dom" has an obvious meaning to you, but it might be prudent to introduce the term to ease the programmer into it. It is a little bit on the side of jargon...  maybe with a link to https://en.wikipedia.org/wiki/Document_Object_Model Interestingly, the wikipedia page does not mention JSON. 

(This is the downside of "document". We know what a document is... not everyone knows what a dom is... even when they know perfectly well what JSON is.)

Reply:
(I just want to stress that "document" is an English word whereas dom is... well... technical jargon... So while I agree with the change, we have to pay attention to the programmers who are not familiar with the concept of a dom.)

Reply:
Hmm, I could imagine `tree` working pretty well as well. That's how most "one shot parsers" are described. Even if we're not storing it that way, that's how it looks ...

Reply:
I'll certainly not start a debate over such naming conventions. Tree would be good too. I found that document was fine too, though I understand why you want to change it.

Reply:
Yeah, not wanting document has a lot to do with it being a class. Though also, "document" has always confused me when I see it. I usually think of a human-edited thing like Microsoft Word, even though I understand we use it in a technical sense to describe a file in a format.

Your concern about dom makes sense though, and I'd rather have something lasting. I'm going to let it stew for a day or two. Meantime, I'm going to merge this, and rename in a couple of days mechanically if we think of one that makes more sense, because a lot of this PR has more to do with "change to using a namespace" and not the specific name of the namespace, and I'd like to build on top of that.